hazelcast/src/test/java/com/hazelcast/client/alto/AltoChannelConnectorTest.java

```diff
@@ -0,0 +1,254 @@
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+import org.mockito.stubbing.OngoingStubbing;
+
+import java.io.IOException;
+import java.util.concurrent.ExecutorService;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+import static org.junit.Assert.assertFalse;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.RETURNS_DEEP_STUBS;
+import static org.mockito.Mockito.doAnswer;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@RunWith(HazelcastParallelClassRunner.class)
+@Category(QuickTest.class)
+public class AltoChannelConnectorTest {
+
+    private AltoChannelConnector connector;
+    private TcpClientConnection mockConnection;
+    private Channel[] mockAltoChannels;
+    private AltoChannelConnector.ChannelCreator mockChannelCreator;
+
+    @Before
+    public void setup() {
+        mockConnection = setupMockConnection();
+        int channelCount = 5;
+        mockAltoChannels = setupMockAltoChannels(channelCount);
+        mockChannelCreator = setupMockChannelCreator(mockAltoChannels);
+        connector = new AltoChannelConnector(UuidUtil.newUnsecureUUID(),
+                mockConnection,
+                IntStream.range(0, channelCount).boxed().collect(Collectors.toList()),
+                setupMockExecutorService(),
+                mock(ChannelInitializer.class),
+                mockChannelCreator,
+                mock(LoggingService.class, RETURNS_DEEP_STUBS));
+    }
+
+    @Test
+    public void testConnector() throws IOException {
+        connector.initiate();
+        verify(mockConnection, times(1)).setAltoChannels(any());
+
+        // We should write authentication bytes to every channel
+        // and do not close them.
+        for (Channel channel : mockAltoChannels) {
+            verify(channel, times(1)).write(any());
+            verify(channel, never()).close();
+        }
+    }
+
+    @Test
+    public void testConnector_whenChannelCreationFails() {
+        doThrow(new RuntimeException("expected"))
+                .when(mockChannelCreator)
+                .create(any(), any(), any());
+
+        connector.initiate();
+
+        // Since no channel is returned from the channel creator,
+        // we can only assert that we have never called setAltoChannels
+        verify(mockConnection, never()).setAltoChannels(any());
+    }
+
+    @Test
+    public void testConnector_whenConnectionIsClosed() {
+        when(mockConnection.isAlive()).thenReturn(false);
+
+        connector.initiate();
+
+        verify(mockConnection, never()).setAltoChannels(any());
+
+        // We should not even attempt to write anything if the connection
```

Comment:
I think it makes sense to assert that we don't even try to create an alto channel